utilities/cli/util.py

```diff
@@ -314,27 +314,89 @@ def check_nvidia_ctk(min_version: str = "1.12.0", recommended_version: str = "1.
         fatal(f"Could not determine nvidia-ctk version. Version {min_version}+ required.")
 
 
+def get_gpu_name() -> Optional[str]:
+    """
+    Helper function to get GPU name from nvidia-smi.  Returns None if nvidia-smi is not available.
+    """
+    if not shutil.which("nvidia-smi"):
+        return None
+    try:
+        output = subprocess.check_output(
+            ["nvidia-smi", "--query-gpu=name", "--format=csv,noheader"],
+            text=True,
+            stderr=subprocess.DEVNULL,
+        )
+        return output.strip() if output else None
+    except (subprocess.CalledProcessError, FileNotFoundError):
+        return None
+
+
 def get_host_gpu() -> str:
     """Determine if running on dGPU or iGPU"""
-    if not shutil.which("nvidia-smi"):
+    gpu_name = get_gpu_name()
+    if gpu_name is None:
         print(
             "Could not find any GPU drivers on host. Defaulting build to target dGPU/CPU stack.",
             file=sys.stderr,
         )
         return "dgpu"
 
-    try:
-        output = subprocess.check_output(
-            ["nvidia-smi", "--query-gpu=name", "--format=csv,noheader"]
-        )
-        if not output or "Orin (nvgpu)" in output.decode():
-            return "igpu"
-    except subprocess.CalledProcessError:
-        return "dgpu"
-
+    # Check for iGPU (Orin integrated GPU)
+    if "Orin (nvgpu)" in gpu_name:
+        return "igpu"
     return "dgpu"
 
 
+def get_default_cuda_version() -> str:
```

Comment:
please confirm this one, I copy it from @agirault script, guess the logic would change in the next few GA releases? @tbirdso